data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1,5 +1,385 @@
+        description= _ "This weapon, when used offensively, deals damage to all units adjacent to the caster when it hits."
+        special_note=_ "This unit has wide-area attacks centered on the caster."
+        active_on = offense
+    [/dummy]
+[/specials]
+[/attack]
+    [event]
+        name = attacker_hits
+        id = nova
+        first_time_only = no
+        [filter_attack]
+            special_id = nova
+        [/filter_attack]
+        [harm_unit]
+            [filter]
+                [not]
+                    find_in = second_unit
+                [/not]
+                [filter_adjacent]
+                    find_in = unit
+                [/filter_adjacent]
+            [/filter]
+            amount = $weapon.damage
+            damage_type = $weapon.type
+            alignment = $unit.alignment
+            kill = yes
+            experience = yes
+            fire_event = yes
+        [/harm_unit]
+    [/event]
+[+attack]
+[+specials]
+# wmlxgettext: [/specials]
+# wmlxgettext: [/attack]
+#enddef
+
+#define WEAPON_SPECIAL_RAY
+# wmlxgettext: [attack]
+# wmlxgettext: [specials]
+    [dummy]
+        id = ray
+        name = _"ray"
+        description= _ "This weapon, when used offensively, also deals damage to a unit behind the target."
+        active_on = offense
+    [/dummy]
+[/specials]
+[/attack]
+    [event]
+        name = attacker_hits
+        id = ray_hits
+        first_time_only = no
+        [filter_attack]
+            special_id = ray
+        [/filter_attack]
+        [harm_unit]
+            [filter]
+                [filter_adjacent]
+                    find_in = second_unit
+                    formula = "
+                        unit_at(direction_from(self.loc,caster.facing)) = other
+                    where
+                        caster = unit_at(direction_from(self.loc, self.facing))
+                    "
+                [/filter_adjacent]
+            [/filter]
+            amount = $weapon.damage
+            damage_type = $weapon.type
+            alignment = $unit.alignment
+            kill = yes
+            experience = yes
+            fire_event = yes
+        [/harm_unit]
+    [/event]
+[+attack]
+[+specials]
+# wmlxgettext: [/specials]
+# wmlxgettext: [/attack]
+#enddef
+
+#define ABILITY_TAILWIND X Y
+    [dummy]
+        id = tailwind_{X}
+        name = _ "tailwind +"+{X}
+        description = _ "At the start of the turn this unit increases movement points of surrounding units by +"+{X}
+    [/dummy] # wmlxgettext: [abilities]
+[/abilities]
+    [event]
+        name = turn refresh
+        first_time_only = no
+        id = tailwind_{X}
+
+        [modify_unit]
+            [filter]
+                side = $side_number
+                [filter_adjacent]
+                    ability = tailwind_{X}
+                    is_enemy = no
+                [/filter_adjacent]
+                [not]
+                    [filter_adjacent]
+                        ability = tailwind_{Y}
```

Comment:
This code is overly complex for something that doesn't generalise.

Because of passing `Y` as a macro argument, all the `[unit_type]`s with `tailwind_1`have to declare that it doesn't stack with `tailwind_2`. The `[unit_type]` with `tailwind_2` has to declare some value for `Y`, and it declares that it doesn't stack with `tailwind_3`.

However, if any unit did have `tailwind_3` then this code would let `tailwind_1` stack with it. So the code might as well be simpler and hardcode that there's only 2 levels of tailwind.

Reply:
Come to think of it, if the goal is just to have the three tailwinds not stack, is there any reason they need a number encoded into the ID at all?

Reply:
It was done this way because without ID as soon as there is unit with tailwind +2 all tailwind will be +2, so mystics and shamans despite having it set to +1 will also give +2 and it doesn't matter if unit with tailwind +2 dies latter because there event is already there. This is also why we had to prevent the stacking.

Reply:
> as soon as there is unit with tailwind +2 all tailwind will be +2, so mystics and shamans despite having it set to +1 will also give +2 and it doesn't matter if unit with tailwind +2 dies latter because there event is already there.

??????

Why would that be the case? Can't the tailwind event somehow read the amount to increase from the unit providing the ability?

The logic would be something like:

* Find all units adjacent to a unit with tailwind.
* For each of those units, collect all the adjacent units with tailwind.
* Select the adjacent unit with the highest tailwind value.
* Apply that tailwind value; then move on to the next unit.

Reply:
Dont ask me, I dont know. Im just saying what is happening in game and explaining why this was done the way it is. I had a memory of this happening but I tested in like hour ago and indeed it works this way.

Reply:
Right, that happens if the event hard-codes the strength but the IDs of the event and ability do not. I'm saying it shouldn't need to hard-code the strength into the event logic _or_ the IDs. See the "explode" ability in #7249 for an example of how this can be done.